app/Http/Controllers/Api/Account/ApiUserController.php

```diff
@@ -30,18 +36,35 @@ public function show(Request $request): UserResource
      * @param Request $request
      * @param int $termId
      *
-     * @return \Illuminate\Http\JsonResponse
+     * @return JsonResponse
      */
     public function get(Request $request, $termId)
     {
-        $userCompliance = auth()->user()->getStatusForCompliance($termId);
+        try {
+            $term = Term::findOrFail($termId);
+        } catch (ModelNotFoundException $e) {
+            return $this->respondNotFound();
+        }
+
+        $termUser = DB::table('term_user')->where('user_id', auth()->user()->id)
+            ->where('account_id', auth()->user()->account_id)
+            ->where('term_id', $term->id)
+            ->first();
 
-        if (! $userCompliance) {
+        if ($termUser) {
+            $data = [
+                'signed' => true,
+                'signed_date' => DateHelper::getTimestamp($termUser->created_at),
+                'ip_address' => $termUser->ip_address,
+                'user' => new UserResource(auth()->user()),
+                'term' => new ComplianceResource($term),
+            ];
```

Comment:
Maybe this data could be transformed in a Resource? It's used 3 times in this file

Reply:
Interesting, thanks!